server.js

```diff
@@ -809,11 +816,89 @@ app.post('/profilemod', function(req, res) {
     });
 });
 
+/**
+ * upload profile pictures
+ */
+app.post('/updateUserPicture', function (req, res) {
+    if (!req.session.user) {
+        return res.redirect('/');
+    }
+
+    var validImageExtensions = ['image/jpeg', 'image/png'];
+    var uploadedFile = req.files.userpicture;
+    if (!uploadedFile || validImageExtensions.indexOf(uploadedFile.mimetype) === -1) {
+        return res.status(400).send(common.getError(6002));
+    }
+
+    var fileName = common.getUUID();
+    var fileExtension = uploadedFile.mimetype.split('/')[1];
+    var fileObject = {
+        fileName: fileName,
+        filePath: vfs.joinPath(common.vfsTree.USERS, req.session.user._id),
+        fileExtension: fileExtension,
+        fileData: uploadedFile.data,
+        filePermissions: common.vfsPermission.PUBLIC,
+        fileCreator: req.session.user._id
+    };
+
+    vfs.writeFile(fileObject, function (err, fileObj) {
+        if (err) {
+            logger.error(err);
+            return res.status(500).send(common.getError(6003));
+        }
+
+        logger.log(common.formatString('Uploaded: {0}', [fileName]));        
+        users.updateUserPicture(req.session.user._id, fileName, function (err, result) {
+            if (err) {
+                logger.error(err);
+                return res.status(500).send(common.getError(6003));
+            }
+            return res.status(200).send('ok');
+        });
+    });
+});
+
+/**
+ * fetch the profile picture
+ */
+app.get('/profilePicture/:pictureId', function (req, res) {
+    if (!req.session.user) {
+        return res.redirect('/');
+    }
+
+    var pictureId = req.params.pictureId;
+    if (pictureId === 'null') {
+        var defaultImagePath = common.formatString('{0}/public/img/{1}', [__dirname, 'logo.png']);
+        return res.sendFile(defaultImagePath, function (err) {
+            if (err) {
+                logger.error(err);
+            }
+        });
+    }
+    vfs.fileExists(pictureId, function (err, fileObj) {
+        if (err) {
+            logger.error(common.formatString('{0} does not exists',[pictureId]));
+            return res.status(400).send(common.getError(9003));
+        }
+
+        if (fileObj.permission !== common.vfsPermission.PUBLIC) {
+            logger.error(common.formatString('Permission denied to access: {0}',[pictureId]));
+            return res.status(403).send(common.getError(9006));
+        }
+
+        return res.sendFile(fileObj.path, function (err) {
+            if (err) {
+                logger.error(err);
```

Comment:
Handle this error

Reply:
this is a different kind of request. There is no handling in the UI for this kind, unless if we want to make a custom handler

Reply:
Nope, all good